### PR TITLE
end-to-end test for JUnit report

### PR DESCRIPTION
**Pull request by vincentml**
This pull request adds end-to-end tests for JUnit report output from XSpec. 

This is [step 2 of the plan](https://github.com/xspec/xspec/pull/188#issuecomment-376410494) as suggested by @AirQuick

### CODE REVIEW

File: test/end-to-end/processor/_normalizer_junit.xsl [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<xsl:stylesheet exclude-result-prefixes="#all" version="2.0"
+    xmlns:local="x-urn:xspec:test:end-to-end:processor:normalizer:local"
+    xmlns:normalizer="x-urn:xspec:test:end-to-end:processor:normalizer"
+    xmlns:util="x-urn:xspec:test:end-to-end:processor:util"
+    xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:xsl="http://www.w3.org/1999/XSL/Transform">
+    
+    <!-- output serialization parameters should be in line with XSPEC_HOME/src/reporter/junit-report.xsl -->
+    <xsl:output name="junit" method="xml" omit-xml-declaration="yes" indent="yes"/>
```
Comment (AirQuick) [inline test/end-to-end/processor/_normalizer_junit.xsl]: I think this `xsl:output` should be just `<xsl:output name="junit"/>`.
`XSPEC_HOME/src/reporter/junit-report.xsl` has `xsl:output`, but it has `name="escaped"` which is not referenced in `junit-report.xsl`. The serialization parameter is thus effectively `<xsl:output/>`.

File: test/end-to-end/cases/expected/schematron-01-result-norm.html [hunk 2]
```diff
@@ -69,4 +69,4 @@ <h2 class="successful">schematron-01<span class="scenario-totals">passed: 3 / pe
          </table>
       </div>
    </body>
-</html>
\ No newline at end of file
+</html>
```
[the remark this answers is not in the document]
Comment (vincentml) [reply test/end-to-end/cases/expected/schematron-01-result-norm.html]: I've reverted this change. When I rebased the branch I had been trying a new text editor and it had an unexpected default setting to automatically reformat whitespace.

**Comment by AirQuick** [comment]
Thanks, @vincentml , I tested as follows. Looks good to me.

On Windows:
1. Remove all the files (except for `.gitignore`) in `cases/expected/`.
1. Run `generate-expected.cmd`.  Verify that Git detects no change.
1. Compare `*-junit.xml` with `*-result.html` in `cases/expected/`. Verify that they are essentially the same. (Just briefly. Haven't verified in depth.)
1. Compare `*-junit.xml` with `*-junit-norm.xml` in `cases/expected/`. Verify that they are essentially the same.
1. Run `run-e2e-tests.cmd`. Verify that all the tests return OK.

On Linux:
1. Remove all the files (except for `.gitignore`) in `cases/expected/`.
1. Run `generate-expected.sh`. Verify that Git detects no change.
1. Run `run-e2e-tests.sh`. Verify that all the tests return OK.

On Windows and Linux:
1. Modify JUnit report by editing `junit-report.xsl`. Run `run-e2e-tests.cmd` (or `sh`). Verify that the test fails and exits with status = 1.
1. Revert the changes made in the previous step.
1. Modify HTML report by editing `format-xspec-report.xsl`. Run `run-e2e-tests.cmd` (or `sh`). Verify that the test fails and exits with status = 1.

**Comment by AirQuick** [comment]
I'll merge this in a few days.
Then I'll fix #211 and submit it with a test utilizing this pull request.

**Comment by AirQuick** [comment]
@cirulls 
Merged this.

**Pull request merged**